sockeye/checkpoint_decoder.py

```diff
@@ -109,44 +114,41 @@ def __init__(self,
                 self.batch_size = sample_size
 
         for i, factor in enumerate(self.inputs_sentences):
-            write_to_file(factor, os.path.join(self.model, C.DECODE_IN_NAME % i))
-        write_to_file(self.target_sentences, os.path.join(self.model, C.DECODE_REF_NAME))
+            write_to_file(factor, os.path.join(model_folder, C.DECODE_IN_NAME % i))
+        write_to_file(self.target_sentences, os.path.join(model_folder, C.DECODE_REF_NAME))
 
         self.inputs_sentences = list(zip(*self.inputs_sentences))  # type: List[List[str]]
 
-        logger.info("Created CheckpointDecoder(max_input_len=%d, beam_size=%d, model=%s, num_sentences=%d, context=%s)",
-                    max_input_len if max_input_len is not None else -1, beam_size, model, len(self.target_sentences),
-                    context)
+        # TODO: possibly support decoding on multiple GPUs
+        context = contexts[0]
+        self.translator = inference.Translator(
+            batch_size=self.batch_size,
+            context=context,
+            ensemble_mode=self.ensemble_mode,
+            length_penalty=inference.LengthPenalty(self.length_penalty_alpha, self.length_penalty_beta),
+            brevity_penalty=inference.BrevityPenalty(weight=0.0),
+            beam_prune=0.0,
+            beam_search_stop='all',
+            nbest_size=self.nbest_size,
+            models=[self.model],
+            source_vocabs=source_vocabs,
+            target_vocab=target_vocab,
+            restrict_lexicon=None,
+            store_beam=False)
```

Comment:
would it make sense to expose hybridize here (and also in the CheckpointDecoder class)?

Reply:
sure, i'll expose this.